src/Apps/Order2/Routes/Checkout/Components/Order2PaymentStep/Order2PaymentForm.tsx

```diff
@@ -0,0 +1,243 @@
+import { Box, Button, Flex, Spacer, Text } from "@artsy/palette"
+import ReceiptIcon from "@artsy/icons/ReceiptIcon"
+import {
+  Elements,
+  PaymentElement,
+  useElements,
+  useStripe,
+} from "@stripe/react-stripe-js"
+import {
+  type StripeElementsOptions,
+  type StripeElementsUpdateOptions,
+  type StripePaymentElementOptions,
+  loadStripe,
+} from "@stripe/stripe-js"
+import { Collapse } from "Apps/Order/Components/Collapse"
+import { FadeInBox } from "Components/FadeInBox"
+import { getENV } from "Utils/getENV"
+import { useState } from "react"
+import LockIcon from "@artsy/icons/LockIcon"
+import { RouterLink } from "System/Components/RouterLink"
+import createLogger from "Utils/logger"
+import { graphql, useRelayEnvironment, fetchQuery } from "react-relay"
+import type { Order2PaymentFormConfirmationTokenQuery } from "__generated__/Order2PaymentFormConfirmationTokenQuery.graphql"
+
+const stripePromise = loadStripe(getENV("STRIPE_PUBLISHABLE_KEY"))
+const logger = createLogger("Order2PaymentForm")
+
+interface Props {
+  order: any
+  setConfirmationToken: (token: any) => void
+}
+
+export const Order2PaymentForm = ({ order, setConfirmationToken }: Props) => {
+  const { itemsTotal, seller } = order
+
+  if (!itemsTotal) {
+    throw new Error("itemsTotal is required")
+  }
+
+  const orderOptions: StripeElementsUpdateOptions = {
+    amount: itemsTotal.minor,
+    currency: itemsTotal.currencyCode.toLowerCase(),
+    onBehalfOf: seller?.merchantAccount?.externalId,
+  }
+
+  const options: StripeElementsOptions = {
+    mode: "payment",
+    appearance: {
+      variables: {
+        accordionItemSpacing: "10px",
+      },
+      rules: {
+        ".AccordionItem": {
+          border: "none",
+          backgroundColor: "#EFEFEF",
+        },
+        ".AccordionItem:focus-visible": {
+          border: "1px solid black",
+        },
+        ".AccordionItem--selected": {
+          border: "1px solid black",
+        },
+      },
+    },
+    ...orderOptions,
+  }
+
+  return (
+    <Elements stripe={stripePromise} options={options}>
+      <PaymentFormContent setConfirmationToken={setConfirmationToken} />
+    </Elements>
+  )
+}
+
+const PaymentFormContent = ({ setConfirmationToken }) => {
+  const stripe = useStripe()
+  const elements = useElements()
+  const environment = useRelayEnvironment()
+
+  if (!(stripe && elements)) {
+    return null
+  }
+
+  // From Stripe docs
+  const [errorMessage, setErrorMessage] = useState()
+
+  // TODO: move this to CheckoutContext
+  const [selectedPaymentMethod, setSelectedPaymentMethod] = useState("new")
```

Comment:
If something is definitely local to a step and has no effect on the wider context it might be fine to leave it scoped to the area it's relevant and avoid all the dispatch/reducer stuff.